probe/awsecs/reporter.go

```diff
@@ -0,0 +1,121 @@
+package awsecs
+
+import (
+	log "github.com/Sirupsen/logrus"
+	"github.com/weaveworks/scope/probe/docker"
+	"github.com/weaveworks/scope/report"
+)
+
+// TaskFamily is the key that stores the task family of an ECS Task
+const (
+	TaskFamily = "ecs_task_family"
+)
+
+type taskInfo struct {
+	containerIDs []string
+	family       string
+}
+
+// return map from cluster to map of task arns to task infos
+func getLabelInfo(rpt report.Report) map[string]map[string]*taskInfo {
+	results := make(map[string]map[string]*taskInfo)
+	log.Debug("scanning for ECS containers")
+	for nodeID, node := range rpt.Container.Nodes {
+
+		taskArn, taskArnOk := node.Latest.Lookup(docker.LabelPrefix + "com.amazonaws.ecs.task-arn")
+		cluster, clusterOk := node.Latest.Lookup(docker.LabelPrefix + "com.amazonaws.ecs.cluster")
+		family, familyOk := node.Latest.Lookup(docker.LabelPrefix + "com.amazonaws.ecs.task-definition-family")
+
+		if taskArnOk && clusterOk && familyOk {
+			taskMap, ok := results[cluster]
+			if !ok {
+				taskMap = make(map[string]*taskInfo)
+				results[cluster] = taskMap
+			}
+
+			task, ok := taskMap[taskArn]
+			if !ok {
+				task = &taskInfo{containerIDs: make([]string, 0), family: family}
+				taskMap[taskArn] = task
+			}
+
+			task.containerIDs = append(task.containerIDs, nodeID)
+		}
+	}
+	log.Debug("Got ECS container info: %v", results)
+	return results
+}
+
+// Reporter implements Tagger
+type Reporter struct {
+}
+
+// Tag needed for Tagger
+func (r Reporter) Tag(rpt report.Report) (report.Report, error) {
+	rpt = rpt.Copy()
+
+	clusterMap := getLabelInfo(rpt)
+
+	for cluster, taskMap := range clusterMap {
+		log.Debugf("Fetching ECS info for cluster %v with %v tasks", cluster, len(taskMap))
+
+		client, err := newClient(cluster)
+		if err != nil {
+			return rpt, err
+		}
+
+		taskArns := make([]string, 0, len(taskMap))
+		for taskArn := range taskMap {
+			taskArns = append(taskArns, taskArn)
+		}
```

Comment:
I'd rather keep the weirdness of AWS's api types contained within that function.
And it's not really a matter of "back to" a `[]*string`, this is just a `keys()` operation on the map. Is there an easier way to get `[]*string` out of a map?